tchannel/tornado/dispatch.py

```diff
@@ -149,14 +147,18 @@ def handle_call(self, request, connection):
         expected_as = handler.req_serializer.name
 
         if request.endpoint in self.handlers and requested_as != expected_as:
-            connection.send_error(
-                ErrorCode.bad_request,
-                "Your serialization was '%s' but the server expected '%s'" % (
-                    requested_as,
-                    expected_as,
+            connection.send_error(BadRequestError(
+                description=(
+                    "Your serialization was '%s' but the server expected '%s'"
```

Comment:
I would just do,

> Server expected a "%s" request but got "%s"

So it'll end up like 'Server expected "thrift" request but got "json"'.